source/common/upstream/upstream_impl.h

```diff
@@ -186,6 +194,7 @@ class HostImpl : public HostDescriptionImpl,
 
 private:
   std::atomic<uint64_t> health_flags_{};
+  std::atomic<uint64_t> health_failure_type_;
```

Comment:
Nit: initialize with `{}` to void undefined state, as with `health_flags_{}` above.